nntrainer/src/fc_layer.cpp

```diff
@@ -200,11 +200,16 @@ Tensor FullyConnectedLayer::backwarding(Tensor derivative, int iteration) {
     }
   }
 
+  bool isL2norm = weight_decay.type == WeightDecayType::l2norm;
```

Comment:
how about add this as a common generic "protected" function in `layer.h`, and any function which has weight can use this?

Reply:
Updated, thanks for the suggestion!

Reply:
I am sorry about unclear comment earlier. I meant applying weight decay - operation and weights.
Independent on the type of weights, the weight decay will be applied in the same way.
IMO, it would be better to write out the operation `.applyIf(this->isWeightDecayL2Norm(), _LIFT(add_i), weight, weight_decay.lambda)` once dependent on the `weight` as an argument.

Reply:
Making a new operation something like `Tensor::applyWeightDecay(weightdecay)` comes to my mind.
Do you have any better idea?